responses/__init__.py

```diff
@@ -305,23 +309,29 @@ def is_closed():
 
 
 class BaseResponse(object):
-    passthrough = False
-    content_type = None
-    headers = None
-    stream = False
+    passthrough: bool = False
+    content_type: Optional[str] = None
+    headers: Optional[Mapping[str, str]] = None
```

Comment:
@markstory 
thing becomes a bit more complecated since we use `HTTPHeaderDict`

it has type:
`class HTTPHeaderDict(MutableMapping[str, str]):`, see https://github.com/urllib3/urllib3/blob/e16beb210c03c6f5ce4e0908bddb6556442b6a37/src/urllib3/_collections.py#L211


then in our implementatino here, we extend it:
https://github.com/getsentry/responses/blob/2a049bdd8fedf1b1a4c23992859ec35194ef43fa/responses/__init__.py#L407

and extend takes either `HTTPHeaderDict`, `Mapping` or `Iterable`:
https://github.com/urllib3/urllib3/blob/e16beb210c03c6f5ce4e0908bddb6556442b6a37/src/urllib3/_collections.py#L334


so, generally list of tuples should be supportes and standard dict as well